consensus/src/block_storage/sync_manager.rs

```diff
@@ -382,9 +424,18 @@ impl BlockRetriever {
                     }
                     peer = self.pick_peer(attempt, peers);
                 }
+                // cover Ok((_, _)) for other BlockRetrievalStatus
+                _ => {
+                    unreachable!()
```

Comment:
no panic when processing external data, this is security vulnerability

Reply:
this path will never run because line 388-395 guarantees only `Ok(BlockRetrievalStatus::SucceededWithEnoughBlocks, _)` or `Ok(BlockRetrievalStatus::SucceededWithTarget, _)` or `Err(_)` are possible.

Reply:
I see, this is confusing. we can directly write
```
response.and_then(|result| {
  match result.status() {
    Ok() | Ok() => {..},
    _ => {},
  }
}.map_err();
```

Reply:
Hmmm, it is easier to `break` the loop from the match than in the closure? I've added some comments.

Reply:
or something like
``` 
match response {
  Ok(result) if matches!(result.status(), BlockRetrievalStatus::Succeeded) => {
  }
  e @ _ => {
  }
}  
```